### PR TITLE
School Library: Unit tests

### PR DESCRIPTION
**Pull request by IndieCoderMM**
# Testing classes with Rspec
We made the following changes in this branch:
- Added tests for `Person` and `Book`
- Added tests for `Student`, `Teacher`, `Classroom`, and `Rental` 
- Added tests for all decorator classes

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    it 'returns the correct age' do
+      expect(@person.age).to eql(20)
+    end
+  end
+
+  describe '#can_use_services?' do
+    it 'should return true' do
+      expect(@person.can_use_services?).to be true
+    end
+  end
```
Comment (Ghiftee) [inline spec/person_spec.rb]: - [ ] Good job adding these tests, however, to ensure you are covering all case scenarios, kindly also check for some positive and negative case scenarios e.g:
```The can_use_services? method returns false if under age and no parent permission```
```The can_use_services? method returns true if over age and no parent permission```

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @IndieCoderMM, @yetemegn-telaye and @Ridwanullahi-code  👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ Unit tests are added for every class
✔️ Tests pass
✔️ Tests have their own ``spec`` directory

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/person_spec.rb (hunk 1 above)
Comment (IndieCoderMM) [reply spec/person_spec.rb]: Thanks for your suggestion. I'll add these tests right away.

**Review by TuyishimireEric — APPROVED** [review]
# Status: APPROVED 🟢  
### Hi  @Team  👋 👋 

#### I'm @TuyishimireEric, Your code reviewer,
Good job on implementing the project requirements! 🥇 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it 🦾  
Congratulations! 🎉

## Highlights
✔ Tests are passing👍 
✔ Made the required changes👍 

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/enus/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
